### PR TITLE
Socket

### PR DESCRIPTION
**Pull request by hugomatic**
And other routes

### CODE REVIEW

File: gz-resources.html [hunk 1]
```diff
@@ -92,126 +121,193 @@
...
         this.$.ajaxgetall.headers.authorization = this.token.trim()
         this.$.ajaxgetall.generateRequest()
+        console.log(this.$.ajaxgetall.method,this.$.ajaxgetall.url, 'request sent')
       },
 
       /// Callback when successfully retrieving all resources
-      _getAllResponse: function(req) {
+      _getAllResponse: function() {
         const resp = this.$.ajaxgetall.lastResponse
+        console.log('_getAllResponse:', resp)
```
Comment (chapulina) [inline gz-resources.html]: Can we remove this log?

File: gz-resources.html [hunk 2]
```diff
@@ -92,126 +121,193 @@
...
 
-      /// Create a resource
-      create: function(req) {
-        this.$.ajaxcreate.headers.authorization = this.token.trim();
-        this.$.ajaxcreate.body = this.data;
-        this.$.ajaxcreate.generateRequest();
+      /// Create a resource from data
+      create: function(data) {
+        // set data
+        this.data = data
```
Comment (chapulina) [inline gz-resources.html:237]: Even if we get `data` here, it gets overwritten inside `_tapCreate` by the contents of `this.datatxt`. How is this function intended to be used now?

Before, external components could use this as follows:
1. Set the `data` variable
2. Call `create()`

And the internal GUI was doing:
1. Input `datatxt`
2. Call `_tapCreate`
3. `datatxt` is passed to `data`
4. `create` is called

File: gz-resources.html [hunk 3]
```diff
@@ -92,126 +124,206 @@
...
       },
 
-      //////////////////////////////////////////////////
       /// Get all
-      //////////////////////////////////////////////////
       /// Get all resources for user
       getAll: function(req) {
         this.$.ajaxgetall.headers.authorization = this.token.trim()
         this.$.ajaxgetall.generateRequest()
+        console.log(this.$.ajaxgetall.method,this.$.ajaxgetall.url, 'request sent')
```
Comment (chapulina) [inline gz-resources.html]: Can we remove this log?

File: gz-resources.html (hunk 2 above)
Comment (hugomatic) [reply gz-resources.html:237]: Sorry for the bug. I fixed this with a common function called from either functions

File: gz-resources.html (hunk 1 above)
Comment (hugomatic) [reply gz-resources.html]: I replaced console.log with this.log, and they are silent by default.

**Comment by chapulina** [comment]
+1

**Comment by chapulina** [comment]
+1

**Pull request merged**